handlers/handlers.go

```diff
@@ -246,11 +249,83 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content string `json:"content"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := &models.Note{
+			AuthorId:     models.UserId(userId),
+			Content:      noteForm.Content,
+			CreationTime: time.Now().UTC(),
+		}
+
+		noteId, err := noteservice.StoreNewNote(note)
+		if err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		responseWriter.Write([]byte(`{NoteId: "` + strconv.FormatInt(int64(noteId), 10) + `"}`))
```

Comment:
is there a reason we can't just return the `noteId` in the content?

Reply:
I'm not sure what you mean by that. The response to this is a json object with the `noteId` of the newly published note. 

are you asking why i had to format the noteid into a string?

Reply:
I'm saying just the `noteId` is a valid json object. We could just return that. If you're not a fan of that we can just return something like `{"id": 3}`. The key should be camelCase and the value shouldn't be in quotes. Additionally, the way I'd expect us to return that is by json serializing some struct